sum_types_generator/lib/src/builder.dart

```diff
@@ -162,8 +162,12 @@ String loadFromRecord(SumTypeSpec spec) => function(
             "${spec.sumTypeName}${specialize(spec.typeParams.map((param) => param.name))}.${expected.name}(",
             if (expected.type.isDirectlyRecursive)
               "load(rec.${expected.name}!)"
-            else if (expected.type.requiresPayload)
-              "rec.${expected.name}!",
+            else if (expected.type.requiresPayload &&
+                expected.parameterStyle == ParameterStyle.positional)
+              "rec.${expected.name}!"
+            else if (expected.type.requiresPayload &&
+                expected.parameterStyle == ParameterStyle.named)
+              "${expected.parameterName}: rec.${expected.name}!",
```

Comment:
The great power of working with enumerations is the exhaustiveness control in `switch`-statements. If later you add a new variant into an `enum` type the compiler will automatically "highlight" (with some diagnostic messages, warnings or errors) all the places in the code where you're `switch`ing on variables of that type, making it very easy to correct the code. Of course, this is only possible, when `switch`-statements are used (and _without_ the `default:` label).

It would be really nice to use a `switch`-statement here, but the context of this "list comprehension" syntax (`[ ... ]`) is limiting, it doesn't allow the use of `switch`-statements. We could sneak one in using the immediately-invoked-function ~pattern~ trick:

```dart
[
  // ...
  if (expected.type.requiresPayload) () {
    switch (expected.parameterStyle) {
      case ParameterStyle.positional: return "rec.${expected.name}!";
      case ParameterStyle.named: return "${expected.parameterName}: rec.${expected.name}!";
    }
  }()
  // ...
]
```

But this may be considered a bit ugly, so, instead, we can extract that closure into a proper function, say, `decorateArg(style, name, value)`. Then the code should look something like this:

```dart
[
  // ...
  if (expected.type.requiresPayload)
    decorateArg(expected.parameterStyle, expected.parameterName, "rec.${expected.name}!"),
  // ...
]
```

where `decorateArg()` would look like this:

```dart
String decorateArg(ParameterStyle parameterStyle, String parameterName, String, argValue) {
    switch (parameterStyle) {
      case ParameterStyle.positional: return argValue;
      case ParameterStyle.named: return "$parameterName: $argValue";
    }
}
```

I'd put this auxiliary function right under the this one (`loadFromRecord()`).